micro-rdk/src/common/webrtc/api.rs

```diff
@@ -283,40 +287,51 @@ impl WebRtcSignalingChannel {
+        // Loop to allow receiving heartbeats without returning the next remote candidate.
+        loop {
+            match self.signaling_rx.next().await {
+                None => {
+                    return Err(WebRtcError::SignalingDisconnected());
+                }
+                Some(req) => {
+                    let req = req?;
+                    if let Some(stage) = req.stage {
+                        match stage {
+                            answer_request::Stage::Update(c) => {
+                                if let Some(c) = c.candidate {
+                                    log::debug!("received candidate {}", c.candidate);
+                                    return c
+                                        .candidate
+                                        .try_into()
+                                        .map_err(|_| WebRtcError::CannotParseCandidate)
+                                        .map(Option::Some);
+                                } else {
+                                    log::error!("received no candidates with this update request");
+                                    return Ok(None);
+                                }
+                            }
+                            answer_request::Stage::Error(s) => {
+                                if let Some(status) = s.status {
+                                    return Err(WebRtcError::SignalingError(status.message));
+                                }
+                                return Err(WebRtcError::SignalingError("unknown".to_owned()));
+                            }
+                            answer_request::Stage::Done(_) => {
                                 return Ok(None);
                             }
-                        }
-                        answer_request::Stage::Error(s) => {
-                            if let Some(status) = s.status {
-                                return Err(WebRtcError::SignalingError(status.message));
+                            answer_request::Stage::Heartbeat(_) => {
+                                log::debug!("received a heartbeat from the signaling server");
+                                continue;
+                            }
+                            _ => {
+                                return Err(WebRtcError::SignalingError(
+                                    "unexpected stage".to_owned(),
+                                ))
                             }
-                            return Err(WebRtcError::SignalingError("unknown".to_owned()));
-                        }
-                        answer_request::Stage::Done(_) => {
-                            return Ok(None);
-                        }
-                        _ => {
-                            return Err(WebRtcError::SignalingError("unexpected stage".to_owned()))
                         }
                     }
+                    return Ok(None);
```

Comment:
not sure if you were aware (and you don't have to make this change), but you can break from loops with return values by declaring `break Ok(None)` for example

Reply:
That's sick, did not know that. I think because the `answer_request::Stage::Done(_)` arm was already doing `return Ok(None)`, I may just use `return Ok(None)` here so there's some consistency.